buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -721,7 +721,12 @@ def set_artifactory_config() {
         for (geo in ARTIFACTORY_CONFIG['geos']) {
             ARTIFACTORY_CONFIG[geo] = [:]
             ARTIFACTORY_CONFIG[geo]['server'] = get_value(VARIABLES.artifactory.server, geo)
-            ARTIFACTORY_CONFIG[geo]['numArtifacts'] = get_value(VARIABLES.artifactory.numArtifacts, geo).toInteger()
+            ARTIFACTORY_CONFIG[geo]['useBuildDiscarderNumArtifacts'] = get_value(VARIABLES.artifactory.useBuildDiscarderNumArtifacts, geo)
+            if (ARTIFACTORY_CONFIG[geo]['useBuildDiscarderNumArtifacts']) {
```

Comment:
Why this condition? Shouldn't be `if (get_value(VARIABLES.artifactory.numArtifacts, geo)) {...}` ?